### DISCUSSION TITLE
NEXTCLOUD_DATADIR= should not require starting with a /

### DISCUSSION THREAD

**Discussion by FlattusBlastus** [nextcloud/all-in-one#2195] [Questions] [answered]
Using docker run, you can define nextcloud_aio_nextcloud_datadir and -e NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir. Using docker compose, - NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir generates an error requesting the input to start with a /.

**Comment by szaimen** [comment, marked as answer]
Hi, you must have made a mistake because as you saw it works with docker run. Also there is an explicit check here: https://github.com/nextcloud/all-in-one/blob/a68b00b126f919639ced90fc3b77a4ac954768c7/Containers/mastercontainer/start.sh#L97

We'll improve the wording with https://github.com/nextcloud/all-in-one/pull/2206/files

**Reply by FlattusBlastus** [reply]
If I blow up my install with NC 26 again, I will get a screenshot for you.

**Reply by FlattusBlastus** [reply]
![image](https://user-images.githubusercontent.com/78336592/227776099-f5eb5eea-2f3b-47ae-8a17-b642ffbeb2b1.png)

**Reply by szaimen** [reply]
You need to remove the quotes around nextcloud_aio_nextcloud_datadir in your docker-compose.

**Reply by FlattusBlastus** [reply]
With " or ' it does this. If neither, it does the startupapps thing.

**Reply by FlattusBlastus** [reply]
version: '3.9'
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /dockerdata/ncmc
  nextcloud_aio_nextcloud_datadir:
    name: nextcloud_aio_nextcloud_datadir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /dockerdata/nextcloud
  nextcloud_aio_backupdir:
    name: nextcloud_aio_backupdir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /dockerbackup
networks:
  nextcloud-aio:
    name: nextcloud-aio
services:
  nextcloud:
    image: nextcloud/all-in-one:beta
    container_name: nextcloud-aio-mastercontainer
    restart: unless-stopped
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - //var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 8080:8080
    environment:
      - APACHE_PORT=11000
      - NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir
      - SKIP_DOMAIN_VALIDATION=true
      - NEXTCLOUD_MEMORY_LIMIT=512M
      - NEXTCLOUD_ADDITIONAL_APKS=imagemagick aria2 yt-dlp ocrmypdf tesseract-ocr-data-deu sane-utils
      - NEXTCLOUD_STARTUP_APPS=twofactor_totp user_usage_report libresign quota_warning deck files_scripts calendar_resource_management calendar_news gestion files_lock memories notes files_zip calendar cloud_py_api group_default_quota notify_push timemanager files_archive camerarawpreviews files_accesscontrol richdocuments recognize tasks user_migration files_3dmodelviewer files_notify_redis groupfolders groupquota analytics dicomviewer previewgenerator workflow_script files_mindmap carnet permissions_overwrite login_notes emlviewer ransomware_protection files_linkeditor files_automatedtagging external side_menu contacts mediadc externalportal integration_youtube files_downloadactivity tvshownamer approval forms sharerenamer geoblocker integration_onedrive files_antivirus files_confidential mail collectives phonetrack files_fulltextsearch music sociallogin appointments cospend fulltextsearch_elasticsearch maps google_synchronization keeweb cadviewer tables event_update_notification ncdownloader flow_notifications files_photospheres announcementcenter fulltextsearch bookmarks unsplash files_bpm registration polls workspace files_fulltextsearch_tesseract metadata workflow_pdf_converter workflow_ocr breezedark webhooks
      - TZ=America/Los_Angeles
      - AUTOMATIC_UPDATES=1
      - NEXTCLOUD_MOUNT=/e
      - NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=imagick inotify
    networks:
      - nextcloud-aio
#you must save and execute this docker-compose.yml in the root of the drive you want the mastercontainer data to live in order for the folder to be created such as d: to make d:/dockerdata/ncmc automatically. You must create your backup folder (d:/dockerbackup) nextcloud data (d:/dockerdata/nextcloud) folders before running compose.

**Reply by szaimen** [reply]
Yeah then the startup_apps is as well incorrectly defined. But datadir works as intended.

**Reply by FlattusBlastus** [reply]
![image](https://user-images.githubusercontent.com/78336592/227776267-42a01d3e-d89f-45f0-8f73-39f426f4133e.png)

**Reply by FlattusBlastus** [reply]
quotes needed for startup apps?

**Reply by FlattusBlastus** [reply]
Can you point the way?

**Reply by FlattusBlastus** [reply]
Two other things... the wiki needs to be reverted for NEXTCLOUD_MOUNT=/e and Windows. It still doesn't work.
Also, please reconsider making this not needed for docker compose installs: #You must create your backup folder (d:/dockerbackup) nextcloud data (d:/dockerdata/nextcloud) folders before running compose.
I am willing to troubleshoot with anyone to get docker compose working correctly.

**Reply by szaimen** [reply]
> quotes needed for startup apps?

I think the digits are the problem. I'll allow them with https://github.com/nextcloud/all-in-one/pull/2210

**Reply by FlattusBlastus** [reply]
I removed the whole startup apps and the mastercontainer was created correctly. Trying to figure out what I'm doing wrong.

**Reply by FlattusBlastus** [reply]
files_3dmodelviewer removed and it works now

**Reply by FlattusBlastus** [reply]
I know you were conducting research on how to create the folders automatically in compose. 

#you must save and execute this docker-compose.yml in the root of the drive you want the mastercontainer data to live in order for the folder to be created such as d: to make d:/dockerdata/ncmc automatically.
In order to support this:
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /dockerdata/ncmc

**Reply by FlattusBlastus** [reply]
/d/dockerdata/ncmc doesn't work by the way.

**Reply by szaimen** [reply]
Did you already try `/host_mnt/d/dockerdata/ncmc` as advised in 
https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
![image](https://user-images.githubusercontent.com/42591237/227777607-9b3c9edc-ffd3-4022-b003-9f420c0c559e.png)

**Reply by FlattusBlastus** [reply]
Yes. I use this before docker compose up -d in order to make it work at all. Hence the requests to add the behavior I observe with the mastercontainer.

docker volume create ^
--driver local ^
--name nextcloud_aio_nextcloud_datadir ^
-o device="/d/dockerdata/nextcloud" ^
-o type="none" ^
-o o="bind"

docker volume create ^
--driver local ^
--name nextcloud_aio_backupdir ^
-o device="/d/dockerbackup" ^
-o type="none" ^
-o o="bind"

**Reply by FlattusBlastus** [reply]
> Did you already try `/host_mnt/d/dockerdata/ncmc` as advised in https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir ![image](https://user-images.githubusercontent.com/42591237/227777607-9b3c9edc-ffd3-4022-b003-9f420c0c559e.png)

So no I haven't. I will. I know that /host_mnt/driveletter works but it is ugly to my Windows guy eyes.
I'll try again when the updater bug is resolved.

**Comment by szaimen** [comment]
This is now improved with v4.8.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
